src/backlinks/handler.js

```diff
@@ -139,13 +139,67 @@ export default async function auditBrokenBacklinks(message, context) {
       auditResult,
     };
 
-    await dataAccess.addAudit(auditData);
+    const audit = await dataAccess.addAudit(auditData);
     const data = {
       type,
       url: site.getBaseURL(),
       auditContext,
       auditResult,
     };
+
+    const opportunities = await dataAccess.Opportunity.allBySiteIdAndStatus(siteId, 'NEW');
+    let brokenBacklinksOppty = opportunities.find((oppty) => oppty.getType() === 'broken-backlinks');
+
+    if (!brokenBacklinksOppty) {
+      const opportunityData = {
+        siteId: site.getId(),
+        auditId: audit.getId(),
+        runbook: 'https://adobe.sharepoint.com/:w:/r/sites/aemsites-engineering/_layouts/15/doc2.aspx?sourcedoc=%7BAC174971-BA97-44A9-9560-90BE6C7CF789%7D&file=Experience_Success_Studio_Broken_Backlinks_Runbook.docx&action=default&mobileredirect=true',
+        type: 'broken-backlinks',
+        origin: 'AUTOMATION',
+        title: 'Authoritative Domains are linking to invalid URLs. This could impact your SEO.',
+        description: 'Provide the correct target URL that each of the broken backlinks should be redirected to.',
+        guidance: {
+          steps: [
+            'Review the list of broken target URLs and the suggested redirects.',
+            'Manually override redirect URLs as needed.',
+            'Copy redirects.',
+            'Paste new entries in your website redirects file.',
+            'Publish the changes.',
+          ],
+        },
+        tags: ['traffic-acquisition'],
+      };
+      brokenBacklinksOppty = await dataAccess.Opportunity.create(opportunityData);
+    } else {
+      brokenBacklinksOppty = brokenBacklinksOppty.setAuditId(audit.getId());
+    }
+
+    if (!data.auditResult.error) {
+      const suggestions = await brokenBacklinksOppty.addSuggestions(
```

Comment:
For recurring weekly audits, I see below issues:

- This will create duplicate entries
- We are not deleting the previous audit suggestions, which didn't surface in the new audit